daemon/protocol/client_lib.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+
+import os
+import re
+import tempfile
+import shutil
+
+from grpc.beta import implementations
+#import grpc
+import daemon_pb2
+
+
+# Copied from daemon/protocol/locate.go
+def _locate_scoot_dir():
+    scootdir = os.environ.get("SCOOTDIR")
+    if scootdir is not None:
+        return scootdir
+    homedir = os.environ.get("HOME")
+    if homedir is not None:
+        return os.path.join(homedir, ".scoot")
+    return None
+
+def _locate_socket():
+    scootdir = _locate_scoot_dir()
+    if scootdir is None:
+        return None
+    return os.path.join(scootdir, "socket")
+
+
+# Singleton variables.
+_domain_sock = _locate_socket()
+_client = None
+_context = None
+
+class ScootException(Exception):
+    """
+    """
+    def __init__(self, caught):
```

Comment:
it'd probably make sense to implement `__repr__` and `__str__` for this as well, so that the repr'd/string representation of the exception clearly shows the wrapped exception. e.g.

```
>>> class ScootException(Exception):
...   def __init__(self, caught):
...     self.caught = caught
... 
>>> e = ScootException(Exception('blah'))
>>> e
ScootException()
```

vs

```
>>> class ScootException(Exception):
...   def __init__(self, caught):
...     self.caught = caught
...   def __repr__(self):
...     return '{}({!r})'.format(self.__class__.__name__, self.caught)
...   
... 
>>> e = ScootException(Exception('blah'))
>>> e
ScootException(Exception('blah',))
```

and `__str__` could just return `repr(self)` - or you could also call `super(Exception, self).__init__(...)` to define the string representation using the base `Exception` mechanics.